### PR TITLE
label Deposit/Submit for Approval button as appropriate

### LINKED ISSUE #544: When review workflow is on, depositor should see a "Submit for Approval" button instead of a "Deposit" button
See Design Doc: https://docs.google.com/document/d/1fcr2DYo7OrX-qTdeUOTWKSS1rlq_47sehESmbvgUsrk/edit#heading=h.vhwtx3wnoqek

"Deposit/Submit for approval button: Is enabled if all 8 sections are green 029B76 (or completed), appears disabled if all sections not complete

- Deposit button for non-approval workflow
- Submit for approval for approval workflow"

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

Fixes #544 - the "Deposit" button for a work should be "Submit for approval" for works in collections with an approval process.

### Before

image

### After

(only for works in collections with review process)

image

## How was this change tested?

unit tests, Amy reviewed it in qa environment.

### CODE REVIEW

File: app/javascript/modules/validate-forms.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Example starter JavaScript for disabling form submissions if there are invalid fields
+// JavaScript to disable form submissions if there are invalid fields
```
Comment (ndushay) [inline app/javascript/modules/validate-forms.js:1]: this comment change is in a separate commit;  I know this file started as a straight up cut and paste from Bootstrap documentation, but ... it's ours now.

File: spec/components/buttons_component_spec.rb [hunk 2]
```diff
@@ -6,12 +6,22 @@
 RSpec.describe ButtonsComponent do
   let(:component) { described_class.new(form: form) }
   let(:form) { ActionView::Helpers::FormBuilder.new(nil, work_form, controller.view_context, {}) }
-  let(:work) { build(:work, collection: build(:collection, id: 7)) }
+  let(:work) { build(:work, collection: create(:collection, id: 7)) }
```
Comment (ndushay) [inline spec/components/buttons_component_spec.rb:9]: given the changes in `buttons_component.rb`, the collection needs to be in the db for the `.find` call

File: app/components/buttons_component.rb [hunk 3]
```diff
@@ -8,4 +8,16 @@ def initialize(form:)
...
+  def submit_button_label
+    work_in_reviewed_coll? ? 'Submit for approval' : 'Deposit'
+  end
+
+  private
+
+  def work_in_reviewed_coll?
+    return false if form.object.instance_of?(CollectionForm)
+
+    Collection.find(form.object.collection_id).review_enabled?
```
Comment (ndushay) [inline app/components/buttons_component.rb:21]: per Mike, there will always be a collection for a work.

Comment (jcoyne) [reply app/components/buttons_component.rb:21]:
```suggestion
    form.object.collection.review_enabled?
```

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/components/buttons_component_spec.rb (hunk 2 above)
Comment (jcoyne) [reply spec/components/buttons_component_spec.rb:9]: with the suggestion I gave above, this approach would not be required.

File: spec/components/works/work_form_component_spec.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@
 RSpec.describe Works::WorkFormComponent do
   let(:component) { described_class.new(work_form: work_form) }
   let(:form) { ActionView::Helpers::FormBuilder.new(nil, work_form, controller.view_context, {}) }
-  let(:work) { build(:work, collection: build(:collection, id: 7)) }
+  let(:work) { build(:work, collection: create(:collection, id: 7)) }
```
Comment (jcoyne) [inline spec/components/works/work_form_component_spec.rb:9]: This also shouldn't need to be persisted.

File: app/components/buttons_component.rb (hunk 3 above)
Comment (ndushay) [reply app/components/buttons_component.rb:21]: I tried this before, and I tried it again just now:

```
     ActionView::Template::Error:
        undefined method `collection' for #<WorkForm:0x00007fb5d028a040>
        Did you mean?  collection_id
      # ./app/components/buttons_component.rb:22:in `work_in_reviewed_coll?'
      # ./app/components/buttons_component.rb:13:in `submit_button_label'
      # ./app/components/buttons_component.html.erb:4:in `call'
```

Comment (jcoyne) [reply app/components/buttons_component.rb:21]: sorry, try `form.object.model.collection`

Comment (ndushay) [reply app/components/buttons_component.rb]: yes - that worked!

**Comment by ndushay** [comment]
@jcoyne - your second suggestion did the job - this is ready for merging!

**Review by jcoyne — APPROVED** [review]

**Pull request merged**